lib/Sema/TypeCheckConcurrency.cpp

```diff
@@ -4814,6 +4815,70 @@ namespace {
   };
 } // end anonymous namespace
 
+/// Compute the isolation of a closure or local function from its parent
+/// isolation.
+///
+/// Doesn't apply preconcurrency because it's generally easier for the
+/// caller to do so.
+static ActorIsolation
+computeClosureIsolationFromParent(DeclContext *closure,
+                                  ActorIsolation parentIsolation,
+                                  bool checkIsolatedCapture) {
+  // We must have parent isolation determined to get here.
+  switch (parentIsolation) {
+  case ActorIsolation::CallerIsolationInheriting:
```

Comment:
I'm surprised that this isn't propagating CallerIsolationInheriting. This isn't new to your PR, though.

Reply:
Yeah, I was thinking about this while working on the patch. I'm not sure if it would be more correct to propagate CallerIsolationInheriting itself or to try to capture to specific isolation from the outer function.